internal/collections/syncmap.go

```diff
@@ -0,0 +1,59 @@
+package collections
```

Comment:
This is here in `core` so that leaf packages like `cachedvfs` don't need to import `core`.